UnityProject/Assets/Scripts/Managers/PlayerInfo.cs

```diff
@@ -52,9 +55,20 @@ public class PlayerInfo
 	public CharacterSheet RequestedCharacterSettings { get; set; }
 
 	/// <summary>The player GameObject. Different GameObject if in lobby vs. in game.</summary>
-	public GameObject GameObject
+	public GameObject GameObject //TODO Better system
 	{
-		get => gameObject;
+		get
+		{
+			if (Mind == null)
+			{
+				return gameObject;
+			}
+			else
+			{
+				return Mind.GetDeepestBody().gameObject;
```

Comment:
what happens if there is no deepest body?

Reply:
rreturns the mind itself